### PR TITLE
#146 GitHub ci docker integration

### LINKED ISSUE #146: Integrate building docker images with GitHub CI
## Expected Behaviour

The CI workflow should include a step to build a docker image for legacy-api and data-service.

## Possible Solution

1. Use https://github.com/docker/build-push-action
2. Apply on push to any branch - tag images with branch name
2. Set up a Docker Hub repo for UC and push images after build(?)

### PR DESCRIPTION
**Pull request by katherine-sussol**
Fixes #146.

## Description

Proof of concept to integrate build and push of docker images as part of ci workflow.

- Builds legacy-api and data-service every push (prerequisite for docker image creation)
- Also builds and pushes images to docker hub 
- 3x test dockerhub repos created (at https://hub.docker.com/orgs/unifiedcodes)
- CI Workflow currently uses my login with a temp access token (that can be revoked by me at any time) to do push (prefer to move this to GitHub secrets later - I think only Admins of openmsupply can access?)
- All are currently tagged with latest - tested tagging with branch name but '#' is not allowed in tag names. Probably need to figure a versioning strategy later with GitHub tags.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [x] Tests OK.

### CODE REVIEW

**Comment by mark-prins** [comment]
> prefer to move this to GitHub secrets later - I think only Admins of openmsupply can access?

I can! let me know what to do, and I can set it up.

File: unified-codes/tools/dockerfiles/data-service.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM unified-codes/dependencies:latest
+FROM unifiedcodes/dependencies:latest
```
Comment (mark-prins) [inline unified-codes/tools/dockerfiles/data-service.Dockerfile:1]: Just curious.. is there a reason for the rename?

**Review by mark-prins — APPROVED** [review]
LGTM!

not sure how to test though 😁  would be good to talk it through when we can.

File: unified-codes/tools/dockerfiles/data-service.Dockerfile (hunk 1 above)
Comment (katherine-sussol) [reply unified-codes/tools/dockerfiles/data-service.Dockerfile:1]: I preferred the hyphenation - but Dockerhub doesn't seem to allow them in the name 😞: 
`Organization name should only contain 4 to 30 letters and/or numbers.`

Comment (mark-prins) [reply unified-codes/tools/dockerfiles/data-service.Dockerfile:1]: ah, I thought it must be something like that! oh well.. it works 😁

**Pull request merged**